external-applications/quarkus-workshop-super-heroes/src/test/java/io/quarkus/ts/openshift/heroes/workshop/HeroesOpenShiftIT.java

```diff
@@ -0,0 +1,175 @@
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+
+import java.net.URL;
+
+import static io.restassured.RestAssured.given;
+import static io.restassured.RestAssured.when;
+import static javax.ws.rs.core.HttpHeaders.ACCEPT;
+import static javax.ws.rs.core.HttpHeaders.CONTENT_TYPE;
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+import static javax.ws.rs.core.Response.Status.*;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@OpenShiftTest
+@ManualDeployment(appName = "quarkus-workshop-hero", httpRoot = "/", knownEndpoint = "/")
+@AdditionalResources("classpath:openjdk-11-rhel7.yaml")
+@AdditionalResources("classpath:heroes-database.yaml")
+@AdditionalResources("classpath:hero.yaml")
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+public class HeroesOpenShiftIT {
+
+    @TestResource
+    private URL url;
+
+    private static String heroId;
+
+    private static final String DEFAULT_NAME = "Test Hero";
+    private static final String UPDATED_NAME = "Updated Test Hero";
+    private static final String DEFAULT_OTHER_NAME = "Other Test Hero Name";
+    private static final String UPDATED_OTHER_NAME = "Updated Other Test Hero Name";
+    private static final String DEFAULT_PICTURE = "harold.png";
+    private static final String UPDATED_PICTURE = "hackerman.png";
+    private static final String DEFAULT_POWERS = "Partakes in this test";
+    private static final String UPDATED_POWERS = "Partakes in update test";
+    private static final int DEFAULT_LEVEL = 42;
+    private static final int UPDATED_LEVEL = 43;
+
+    @Test
+    public void testHello() {
+        when()
+                .get(url + "/api/heroes/hello")
+        .then()
+                .statusCode(OK.getStatusCode())
+                .body(is("hello"));
+    }
+
+    @Test
+    public void testOpenApi() {
+        given()
+                .header(ACCEPT, APPLICATION_JSON)
+                .when().get(url + "/openapi")
+                .then()
+                .statusCode(OK.getStatusCode());
+    }
+
+    @Test
+    public void testLiveness() {
+        given()
+                .header(ACCEPT, APPLICATION_JSON)
+                .when().get(url + "/health/live")
+                .then()
+                .statusCode(OK.getStatusCode());
+    }
+
+    @Test
+    public void testReadiness() {
+        given()
+                .header(ACCEPT, APPLICATION_JSON)
+                .when().get(url + "/health/ready")
+                .then()
+                .statusCode(OK.getStatusCode());
+    }
+
+    @Test
+    public void testMetrics() {
+        given()
+                .header(ACCEPT, APPLICATION_JSON)
+                .when().get(url + "/metrics/application")
+                .then()
+                .statusCode(OK.getStatusCode());
```

Comment:
can you do check on certain metrics being exposed?

Reply:
Check done at:

- https://github.com/quarkus-qe/quarkus-openshift-test-suite/pull/28/files#diff-e70bcb24b82b4e87bcce957bf5b39d16R171
- https://github.com/quarkus-qe/quarkus-openshift-test-suite/pull/28/files#diff-424ef8a617a8061b706b69289202f2fbR172